modules/node_groups/README.md

```diff
@@ -0,0 +1,55 @@
+# eks `node_groups` submodule
+
+Helper submodule to create and manage resources related to `eks_node_groups`.
+
+## Assumptions
+* Designed for use by the parent module and not directly by end users
+
+## Node Groups' IAM Role
+The role ARN specified in `var.default_iam_role_arn` will be used by default. In a simple configuration this will be the worker role created by the parent module.
+
+`iam_role_arn` must be specified in either `var.node_groups_defaults` or `var.node_groups` if the default parent IAM role is not being created for whatever reason, for example if `manage_worker_iam_resources` is set to false in the parent.
+
+## `node_groups` and `node_groups_defaults` keys
+`node_groups_defaults` is a map that can take the below keys. Values will be used if not specified in individual node groups.
+
+`node_groups` is a map of maps. Key of first level will be used as unique value for `for_each` resources and in the `aws_eks_node_group` name. Inner map can take the below values.
```

Comment:
I have mixed feelings about this: On the one hand it more closely mirrors the output they'll be given, on the other it is a lot more confusing than just having a list of maps with the note that "`base_name` is a required value and must be unique for each node group". What are your thoughts?

Reply:
Terraform often does stupid things with lists. You can see it when changing the length of the `workers_groups` list and suddenly all the workers are scheduled for re-creation.

We should be going the other way and converting old inputs from lists to maps with for_each.

Having this as a map makes it pretty obvious what the key on created resources will be. It's a much more natural flow than converting from a list to a map using a subkey.